testsuite/mpitests/test_connect_array_fixed_indegree_mpi.sli

```diff
@@ -0,0 +1,117 @@
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+
+/* BeginDocumentation
+   Name: testsuite::test_connect_array_fixed_indegree_mpi - SLI script to test connection with parameter arrays and connection rule fixed_indegree, with hybrid MPI and thread parallelism
+
+   Synopsis: (test_connect_array_fixed_indegree_mpi) run
+
+   Description:
+
+   Tests with hybrid MPI and thread parallelism the possibility to use
+   parameter arrays to define connection weights and/or delays in conjunction
+   with the connection rules fixed_indegree
+
+   FirstVersion: August 2016
+   Author: Bruno Golosio
+   SeeAlso: Connect
+ */
+
+(unittest) run
+/unittest using
+
+is_threaded not { exit_test_gracefully } if
+
+[2 3] % check for 2, 3 processes
+{
+
+/N 20 def          % number of neurons in each subnet
+/K 5 def           % number of connections per neuron
+/VP_per_MPI 2 def  % number of virtual processes per MPI process
+
+ResetKernel
+NumProcesses /num_mpi_procs Set  % gets number of MPI processes
+
+0 <<
+     /total_num_virtual_procs num_mpi_procs VP_per_MPI mul
+>> SetStatus  % sets number of virtual processes
+
+/gid1 N 1 add def            % gid1 = N + 1
+/gid2 N 2 mul def            % gid2 = N * 2
+/net1 [1 N] Range def        % global ids of source subnet
+/net2 [gid1 gid2] Range def  % global ids of target subnet
+
+/iaf_neuron N 2 mul Create ;  % creates source and target subnets 
+
+/NC N K mul def     % number of connections
+/Dmax NC 1 add def  % maximum delay
+
+/Warr [1 NC] Range cv_dv def    % weight array Warr=[1, ..., NC]
+/Darr [2 Dmax] Range cv_dv def  % delay array  Darr=[2, ..., NC+1]
+% cv_dv converts sli-array type to double-vector type
+
+% synapses and connection dictionaries
+/syn_dict << /model /static_synapse /weight Warr /delay Darr >> def
+/conn_dict << /rule /fixed_indegree /indegree K >> def
+
+net1 net2 conn_dict syn_dict Connect % connects source to target subnet
+
+/n_loc 0 def  % used to check n. of neurons belonging to this MPI process
+net2 % loop on all neurons of target subnet
+{
+        /neuron2 exch def
+
+        neuron2 GetStatus /local get  % checks if neuron2 is local or not
+        {
+                /n_loc n_loc 1 add def  % increases n_loc
+                % gets all connections to the target neuron
+                /conn << /target [neuron2] >> GetConnections def
+
+                /sdict_lst conn GetSynapseStatus def % gets info on synapses
+
+                /Warr1 0 array def % creates empty weight array
+
+                sdict_lst % loop on synapses that connect to target neuron
+                {
+                        /sdict exch def
+                        /w sdict /weight get def  % gets synaptic weight
+                        /d sdict /delay get def   % gets synaptic delay
+                        /diff d w sub def         % diff = delay - weight 
+                        diff 1.0 eq assert_or_die   % checks that d = w + 1
```

Comment:
You should avoid `assert_or_die` in MPI tests. Rather, set a boolean flag, e.g., `pass` here indicating a problem. This also applies to all following `assert_or_die`. Then, at the very end, have `pass n_loc 0 gt and`